src/Worker/ReschedulingDecorator.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neighborhoods\KojoWorkerDecoratorComponent\Worker;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use LogicException;
+use Neighborhoods\KojoWorkerDecoratorComponent\WorkerInterface;
+use Neighborhoods\ThrowableDiagnosticComponent\ThrowableDiagnostic;
+use OutOfBoundsException;
+use Throwable;
+
+final class ReschedulingDecorator implements ReschedulingDecoratorInterface
+{
+    use DecoratorTrait;
+    use ThrowableDiagnostic\Builder\Factory\AwareTrait;
+
+    public const MAX_RESCHEDULE_DELAY_SECONDS = 86400;// (24 hours)
+
+    private /*int*/ $rescheduleDelaySeconds;
+    private /*string*/ $jobTypeCode;
+    private /*Connection*/ $connection;
+
+    public function work(): WorkerInterface
+    {
+        $this->runWorker();
+
+        if ($this->getApiV1WorkerService()->isRequestApplied()) {
+            $this->getApiV1WorkerService()
+                ->getLogger()
+                ->warning('Kojo worker state request already applied, not rescheduling.');
+        } else {
+            $this->reschedule();
+        }
+
+        return $this;
+    }
+
+    private function reschedule(): ReschedulingDecoratorInterface
+    {
+        $this->getConnection()->beginTransaction();
```

Comment:
**blocker**

Using `\Doctrine\DBAL\Connection::beginTransaction()` here instead of `\PDO::beginTransaction()` means that there is internal state in userspace (transaction nesting, savepoints, etc.) that kojo does not have access to (if you're interested, you can check out the doctrine code [here](https://github.com/doctrine/dbal/blob/4.0.x/src/Driver/Connection.php) and [here](https://github.com/doctrine/dbal/blob/4.0.x/src/Connection.php) to get a sense of what I'm talking about). It's definitely a pit of failure (though honestly there's not much we can do about it from kojospace), but the simplest way to go about it is to manage transactions at the `\PDO` level (since kojo has a reference to the same `\PDO`, all information is shared across the userspace/kojospace boundary). In LS we go about this by extracting the `\PDO` via `\Doctrine\DBAL\Connection::getWrappedConnection()`, and performing the same `beginTransaction()`, `commit()`, and `rollBack()` operations there

This is a weird and subtle problem, if you're not understanding what I'm talking about, I can put together some examples to demonstrate it (they would be a useful reference to point to in the future anyway)

Reply:
I see what you mean.
I'll make this `PDO` aware instead of `Doctrine\DBAL\Connection` aware.

Reply:
Updates have been made.